File: FacadeCore/IClientService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using BOCore;

namespace FacadeCore
{
    public interface IClientService
    {
        // ClIENT ACCOUNTS
        GetResponse<ClientAccountBO> GetClientAccountById(int id);
        GetResponse<ClientAccountBO> GetClientAccountByIds(List<int> ids);
        GetResponse<ClientAccountWithUnitsBO> GetClientAccountByIdWithUnits(int id);
        GetResponse<SelectBO> GetClientAccountsForSearchList(string searchterm);
        string GetClientAccountNameById(int id);
        string GetClientAccountUnitsNameById(int id);
        GetResponse<ClientAccountBO> GetClientAccountsByProjectId(int id);
        GetResponse<ClientAccountWithUnitsBO> GetClientAccountsByProjectIdWithUnits(int id);
        GetResponse<IdNameBO> GetClientAccountsByProjectIdForSelect(int projectid);
        GetResponse<IdNameBO> GetClientAccountsByProjectIdLast5(int projectid);
        GetResponse<ClientAccountBO> GetClientAccountsByUnitIds(List<int> Unitids);
        GetResponse<ClientAccountBO> GetClientAccountsByDateDeedofSale();
        Response InsertUpdate(ClientAccountBO company);
        Response AddClientAccountToUnit(int unitId, int accountId);
        Response Delete(List<int> ids);
        Response Delete(List<ClientAccountBO> bos);

        // CLIENT CONTACTS
        GetResponse<ClientContactBO> GetClientContactById(int id);
        GetResponse<decimal> GetMaxOwnerPercentage(int accountid, int ownerid);
        Response InsertUpdateClientContact(ClientContactBO company);
        Response DeleteClientContact(List<int> ids);
        Response DeleteClientContact(List<ClientContactBO> bos);

        // CLIENT OWNER TYPE
        GetResponse<ClientOwnerTypeBO> GetClientOwnerTypeById(int id);
        GetResponse<ClientOwnerTypeBO> GetOwnerTypes();
        GetResponse<IdNameBO> GetOwnerTypesForSelect();
        Response InsertUpdateClientOwnerType(ClientOwnerTypeBO company);
        Response DeleteClientOwnerType(List<int> ids);
        Response DeleteClientOwnerType(List<ClientOwnerTypeBO> bos);

        // ClIENT GIFT
        GetResponse<ClientGiftBO> GetClientGiftById(int id);
        GetResponse<ClientGiftBO> GetClientGiftByAccountId(int id);
        GetResponse<ClientGiftWithAccountDetailsBO> GetClientsGifts(int projectid);
        GetResponse<ClientGiftWithAccountDetailsBO> GetClientsGifts(int projectid, List<int> activities);
        Response InsertUpdateClientGift(ClientGiftBO company);
        Response DeleteClientGift(List<int> ids);
        Response DeleteClientGift(List<ClientGiftBO> bos);

        // ClIENT POINT OF ATTENTION (POA)
        GetResponse<ClientPoaBO> GetClientPoaById(int id);
        GetResponse<ClientPoaBO> GetClientPoaByAccountId(int id);
        GetResponse<ClientPoaWithAccountDetailsBO> GetClientsPoas(int projectid);
        GetResponse<ClientPoaWithAccountDetailsBO> GetClientsPoas(int projectid, List<int> activities);
        Response InsertUpdateClientPoa(ClientPoaBO company);
        Response DeleteClientPoa(List<int> ids);
        Response DeleteClientPoa(List<ClientPoaBO> bos);
    }
}

File: Facade/IClientService.vb
Imports BO
Public Interface IClientService

    'ClIENT ACCOUNTS
    Function GetClientAccountById(id As Integer) As GetResponse(Of ClientAccountBO)
    Function GetClientAccountByIds(ids As List(Of Integer)) As GetResponse(Of ClientAccountBO)
    Function GetClientAccountsByIdWithUnits(id As Integer) As GetResponse(Of ClientAccountWithUnitsBO)
    Function GetClientAccountsForSearchList(searchterm As String) As GetResponse(Of SelectBO)
    Function GetClientAccountNameById(id As Integer) As String
    Function GetClientAccountUnitsNameById(id As Integer) As String
    Function GetClientAccountsByProjectId(id As Integer) As GetResponse(Of ClientAccountBO)
    Function GetClientAccountsByProjectIdWithUnits(id As Integer) As GetResponse(Of ClientAccountWithUnitsBO)
    Function GetClientAccountsByProjectIdForSelect(projectid As Integer) As GetResponse(Of IdNameBO)
    Function GetClientAccountsByProjectIdLast5(projectid As Integer) As GetResponse(Of IdNameBO)
    Function GetClientAccountsByUnitIds(Unitids As List(Of Integer)) As GetResponse(Of ClientAccountBO)
    Function GetClientAccountsByDateDeedofSale() As GetResponse(Of ClientAccountBO)
    Function InsertUpdate(company As ClientAccountBO) As Response
    Function AddClientAccountToUnit(unitId As Integer, accountId As Integer) As Response
    Function Delete(ids As List(Of Integer)) As Response
    Function Delete(bos As List(Of ClientAccountBO)) As Response

    'CLIENT CONTACTS
    Function GetClientContactById(id As Integer) As GetResponse(Of ClientContactBO)
    Function GetMaxOwnerPercentage(accountid As Integer, ownerid As Integer) As GetResponse(Of Decimal)
    Function InsertUpdateClientContact(company As ClientContactBO) As Response
    Function DeleteClientContact(ids As List(Of Integer)) As Response
    Function DeleteClientContact(bos As List(Of ClientContactBO)) As Response

    'CLIENT OWNER TYPE
    Function GetClientOwnerTypeById(id As Integer) As GetResponse(Of ClientOwnerTypeBO)
    Function GetOwnerTypes() As GetResponse(Of ClientOwnerTypeBO)
    Function GetOwnerTypesForSelect() As GetResponse(Of IdNameBO)
    Function InsertUpdateClientOwnerType(company As ClientOwnerTypeBO) As Response
    Function DeleteClientOwnerType(ids As List(Of Integer)) As Response
    Function DeleteClientOwnerType(bos As List(Of ClientOwnerTypeBO)) As Response

    'ClIENT GIFT
    Function GetClientGiftById(id As Integer) As GetResponse(Of ClientGiftBO)
    Function GetClientGiftByAccountId(id As Integer) As GetResponse(Of ClientGiftBO)
    Function GetClientsGifts(projectid As Integer) As GetResponse(Of ClientGiftWithAccountDetailsBO)
    Function GetClientsGifts(projectid As Integer, activities As List(Of Integer)) As GetResponse(Of ClientGiftWithAccountDetailsBO)
    Function InsertUpdateClientGift(company As ClientGiftBO) As Response
    Function DeleteClientGift(ids As List(Of Integer)) As Response
    Function DeleteClientGift(bos As List(Of ClientGiftBO)) As Response

    'ClIENT POINT OF ATTENTION (POA)
    Function GetClientPoaById(id As Integer) As GetResponse(Of ClientPoaBO)
    Function GetClientPoaByAccountId(id As Integer) As GetResponse(Of ClientPoaBO)
    Function GetClientsPoas(projectid As Integer) As GetResponse(Of ClientPoaWithAccountDetailsBO)
    Function GetClientsPoas(projectid As Integer, activities As List(Of Integer)) As GetResponse(Of ClientPoaWithAccountDetailsBO)
    Function InsertUpdateClientPoa(company As ClientPoaBO) As Response
    Function DeleteClientPoa(ids As List(Of Integer)) As Response
    Function DeleteClientPoa(bos As List(Of ClientPoaBO)) As Response



End Interface
